scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -317,13 +327,19 @@ public DTLSConnector(final DtlsConnectorConfig configuration, final SessionCache
 	 * @param configuration The configuration options.
 	 * @param connectionStore The registry to use for managing connections to peers.
 	 * @throws NullPointerException if any of the parameters is <code>null</code>.
+	 * @throws IllegalArgumentException if the connection store uses a different cid length.
 	 */
 	protected DTLSConnector(final DtlsConnectorConfig configuration, final ResumptionSupportingConnectionStore connectionStore) {
 		if (configuration == null) {
 			throw new NullPointerException("Configuration must not be null");
 		} else if (connectionStore == null) {
 			throw new NullPointerException("Connection store must not be null");
 		} else {
+			Integer cidLength = configuration.getConnectionIdLength();
+			if (cidLength != null && cidLength > 0 && cidLength != connectionStore.getConnectionIdLength()) {
+				throw new IllegalArgumentException("Connection store must use the same cid length! " + cidLength
+						+ " != " + connectionStore.getConnectionIdLength());
+			}
```

Comment:
> but also in `connectionStore` (in case of custom store) ?

I assume, there will be only rare cases, when a custom store is used. But if, the same cid length must be used as for the `Record.fromByteArray`. I would prefer to keep the configuration values in the `DtlsConnectorConfig`(at least as leading definition ).

So together:
- configuration value in `DtlsConnectorConfig`
- rarely use custom store
I think,  add it to the javadoc of the DTLSConnector constructor will do it best.

Reply:
Just in case there is misunderstanding, I would keep it `DtlsconnectorConfig` but remove it from `connectionStore`.

Reply:
Remove what? The method `getConnectionIdLength`? Or the field `cidLength`? 
My assumption was, that a custom connection store is rarely used, so I don't think, that spending time into that will payout.

Reply:
like express above : 

>I see cidlenght is only needed to newConnectionId which is only called by put which is only called by DTLSconnector ?

>Maybe we can change the API with something like : boolean put(Connection connection, integer cidlenght);

Reply:
The constructor also calls newConnectionId, if a ClientSessionCache is restored.
If you really feel, it's worth, then introduce the DtlsConnectorConfig into the parameters of the constructor and replace the others. But please, shift that into an other PR :-).

Reply:
I missed it was used in the constructor too.
So not so easy to change it.